src/forge/controller/provisioner.py

```diff
@@ -131,81 +158,87 @@ async def get_proc_mesh(
         with_gpus: bool = False,
         num_hosts: int | None = None,
         mesh_name: Optional[str] = None,
+        host_mesh: HostMesh | None = None,
+        env_vars: dict[str, str] | None = None,
     ):
         """Gets a proc mesh.
 
         num_hosts = None implies that you want a local allocation, this may change.
 
         """
+        if env_vars is None:
+            env_vars = {}
+
         async with self._lock:
             server_name = None
             if num_hosts is not None and num_hosts > 0:
                 created_hosts = len(self._server_names)
-                host_mesh, server_name = await self.create_host_mesh(
-                    name=mesh_name,
-                    num_hosts=num_hosts,
-                )
-                host_id = uuid.uuid1()
-                gpu_manager = GpuManager()
-                self._host_gpu_map[host_id] = gpu_manager
-                host_mesh._host_id = host_id
+                if mesh_name is None:
+                    mesh_name = f"alloc_{created_hosts}"
+                if host_mesh is None:
+                    host_mesh, server_name = await self.create_host_mesh(
+                        name=mesh_name,
+                        num_hosts=num_hosts,
+                    )
+                    host_id = uuid.uuid1()
+                    gpu_manager = GpuManager()
+                    self._host_gpu_map[host_id] = gpu_manager
+                    host_mesh._host_id = host_id
+                else:
+                    host_id = host_mesh._host_id
```

Comment:
Another noob q: where is this field defined? Took a quick look in Monarch and couldn't find it